src/olympia/ratings/views.py

```diff
@@ -392,8 +392,34 @@ def get_paginated_response(self, data):
             if show_grouped_ratings and self.get_addon_object():
                 response.data['grouped_ratings'] = dict(GroupedRating.get(
                     self.addon_object.id))
+        if 'addon' in self.request.GET and is_gate_active(
+                self.request, 'ratings-can_reply'):
+            response.data['can_reply'] = (
+                bool(self.request.user.is_authenticated) and
+                self.check_can_reply_permission_for_ratings_list())
         return response
 
+    def check_can_reply_permission_for_ratings_list(self):
+        """Check whether or not the current request contains an user that can
+        reply to ratings we're about to return.
+
+        Used to populate the `can_reply` property in ratings list, when an
+        addon is passed."""
+        # Clone the current viewset, but change permission_classes.
+        viewset = self.__class__(**self.__dict__)
+        viewset.permission_classes = self.reply_permission_classes
+
+        # Create a fake rating with the addon object attached, to be passed to
+        # check_object_permissions().
+        dummy_rating = Rating(addon=self.get_addon_object())
+
+        try:
+            viewset.check_permissions(self.request)
+            viewset.check_object_permissions(self.request, dummy_rating)
```

Comment:
Could just instantiate the `CanDeleteRatingPermission` directly and test with it rather than cloning the viewset and tinkering with the permission classes.  It's still hacky, but could be considered cleaner... I dunno.

Reply:
It's not `CanDeleteRatingPermission` though (I moved it as a drive-by cleanup, but that's all), it's `[AnyOf(GroupPermission(amo.permissions.ADDONS_EDIT), AllowRelatedObjectPermissions('addon', [AllowAddonAuthor]))]` ...

Reply:
Oh.  I got confused by the drive-by clean up at the top of the commit.